lib/identity_cache/query_api.rb

```diff
@@ -468,7 +485,23 @@ def prefetch_one_association(association, records)
             prefetch_embedded_association(records, association, details)
             parent_records = records.map(&details[:cached_accessor_name].to_sym).compact
           else
-            raise ArgumentError.new("Non-embedded has_one associations do not support prefetching yet.")
+            ids_to_parent_record = records.each_with_object({}) do |record, hash|
+              child_ids = Array(record.send(details[:cached_id_name]))
+              child_ids.each do |child_id|
+                hash[child_id] = record
+              end
+            end
+
+            parent_record_to_child_records = Hash.new { |h, k| h[k] = [] }
```

Comment:
Shouldn't the parent just map to a single child record?  If there are multiple then I think active record just uses the first one.  So I think we can just use a `parent_record_to_child_record = {}` hash and use `parent_record_to_child_record[parent_record] ||= child_record` to assign only one record to the parent.